ansible/roles/images/tasks/load.yaml

```
@@ -9,12 +9,32 @@
   register: images_cache_find_result
   changed_when: false
 
-- name: import images from files in the local images cache  # noqa no-changed-when
-  shell:
-    executable: /bin/bash
-    cmd: ctr --address {{ containerd_cri_socket }} --namespace k8s.io images import <(gzip --decompress --stdout {{ item }})
+- block:
+    - name: import images from files in the local images cache # noqa no-changed-when
+      shell:
+        executable: /bin/bash
+        cmd: ctr --address {{ containerd_cri_socket }} --namespace k8s.io images import <(gzip --decompress --stdout {{ item }})
+      when:
+        - item.split('/')[-1] | regex_search('^1\.(\d)+\.(\d)+_images\.tar\.gz$')
+        - item.split('/')[-1].split('.')[1] |int < 24
+      register: ctr_images_import
+      failed_when: ctr_images_import.rc != 0
+      changed_when: ctr_images_import.rc == 0
+      with_items: "{{  images_cache_find_result.files | map(attribute='path') | list }}"
+
+    - name: import images from files to local images cache with mindthegap
+      shell:
+        executable: /bin/bash
+        cmd: mindthegap import image-bundle --image-bundle={{ item }}
+      when:
+        - item.split('/')[-1] | regex_search('^1\.(\d)+\.(\d)+_images\.tar\.gz$')
+        - item.split('/')[-1].split('.')[1] |int >= 24
+      register: mindthegap_images_import
+      failed_when: mindthegap_images_import.rc != 0
+      changed_when: mindthegap_images_import.rc == 0
+      with_items: "{{  images_cache_find_result.files | map(attribute='path') | list }}"
+
+    - name: remove mindthegap if present
+      command: rm /usr/local/bin/mindthegap || true
```

Why do we need to remove it? Keeping it on the host will make the runs a little faster since it won't need to recopy if there are no changes.

```suggestion
    - name: remove mindthegap if present
      file:
        path: "{{ mindthegap_binary_location_on_remote }}"
        state: absent
```

---

IMHO, we should not leave leftover binaries (not only mindthegap) in the final image for reasons of security and minimize the size of the image. Some environments where network copy is not as good as S3 (like vsphere), it takes time to clone/copy AMI/image to each VM.